core/Makefile

```diff
@@ -352,3 +352,6 @@ coverage:  ## generate coverage report
 
 unused:  ## find unused micropython code
 	vulture src src/_vulture_ignore.txt --exclude "messages.py,*/enums/*"
+
+qstr_check: ## check the consistency of QSTR definitions
+	./tools/qstr_checks.py
```

Comment:
Something like going through the whole Rust codebase, finding all the used `MP_QSTR_XXX` values, and sorting them alphabetically? Could be viable.

One disadvantage it would have is no longer grouping them "logically", e.g. we have `// protobuf` at the top with 5 messages

Reply:
yes; the logical grouping isn't really useful for anything

Reply:
Done in [2d3a0aa](https://github.com/trezor/trezor-firmware/pull/2803/commits/2d3a0aa2bba3f73653c3c01e66258b82b1f15530)

Reply:
Can you turn the script into mako template? Then it wouldn't need new Makefile targets.

Reply:
Making it a template using current automation would mean adding the QSTR-generating logic into `common/tools/cointool.py::render`, which handles the `mako` templates.

That does not seem to me like a good place - `cointool.py` has nothing to do with `Rust`.

We might theoretically setup some other template automation more connected with `core`

Reply:
You can put all the logic directly in the mako file (example: `all_modules.py.mako`), no?

Reply:
Correct, very good idea. Done in [5fbf7ee](https://github.com/trezor/trezor-firmware/pull/2803/commits/5fbf7ee937f432a19269e9bd231aef6f27313845)